numpy/lib/npyio.py

```diff
@@ -1179,21 +1212,51 @@ def savetxt(fname, X, fmt='%.18e', delimiter=' ', newline='\n', header='',
         fmt = asstr(fmt)
     delimiter = asstr(delimiter)
 
+    class WriteWrap(object):
+        """ convert to unicode in py2 or to bytes on bytestream inputs """
+        def __init__(self, fh, encoding):
+            self.fh = fh
+            self.encoding = encoding if encoding else 'latin1'
```

Comment:
I don't think this class should set that default; maybe rather do it in the call to it?